### PR TITLE
TEC-4070: Componets top bar menu

### PR DESCRIPTION
**Pull request by samuelvermeuln**
## Task link
[JIRA](https://evinobr.atlassian.net/jira/software/projects/TEC/boards/64?selectedIssue=TEC-4070&text=%5Bfront%5D)

## Task description
Componets top bar 

## How to test
 [Storybook](https://64e4f08113329cd99216030d-noqbhudtyq.chromatic.com/)

## Ambient to test
`chromatic`

### CODE REVIEW

File: src/components/Button/index.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    background: ${(props) =>
+      (props?.theme &&
+        Colors[props.theme as keyof ThemeColors]?.neutral?.neutral400) ||
+      ''};
+    color: ${(props) =>
+      (props?.theme &&
+        Colors[props.theme as keyof ThemeColors]?.neutral?.neutral100) ||
+      ''};
+  }
+}`
```
Comment (joelsalmeida) [inline src/components/Button/index.tsx:90]: This bracket "}" shouldn't be here.

File: src/components/Button/index.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    ''};
+
+
+  overflow: hidden;
+  box-shadow: none;
+  cursor: pointer;
+  text-decoration: none;
+  border-width: 0;
+  transition:
+    background 0.3s,
```
Comment (joelsalmeida) [inline src/components/Button/index.tsx:63]: Indentation.

File: src/types/index.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export type Severity = 'info' | 'warning' | 'error'
+export type ThemeType = 'light' | 'dark'
+export type Position = 'top-start' | 'top-end' | 'bottom-start' | 'bottom-end'
+
+export interface Props {
```
Comment (joelsalmeida) [inline src/types/index.ts]: "Props" is a too generic name.

**Comment by joelsalmeida** [comment]
Sugestion. Some components are being created with "function" and others with "arrow function", it would be nice to establish a standard and add it as a lint rule.

File: src/components/TopBar/index.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+interface CepProps {
+  cepText?: string
+}
+
+export interface TopBarProps {
+  cepText?: string
+  tagValue?: string
+  iconTag?: React.ReactNode
+  infoText1?: string
```
Comment (joelsalmeida) [inline src/components/TopBar/index.tsx]: Shouldn't this be an array of strings?

File: src/types/index.ts (hunk 3 above)
Comment (samuelvermeuln) [reply src/types/index.ts]: This is to be generic even though it is used in several places

File: src/components/TopBar/index.tsx (hunk 4 above)
Comment (samuelvermeuln) [reply src/components/TopBar/index.tsx]: no

File: src/components/Button/index.tsx (hunk 1 above)
Comment (samuelvermeuln) [reply src/components/Button/index.tsx:90]: he is in the right place

Comment (gabrielcdiniz) [reply src/components/Button/index.tsx:90]: maybe needs remove the bracket in the line 90. Isn't seems right

File: src/components/Button/index.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        Colors[props.theme as keyof ThemeColors]?.neutral?.neutral400) ||
+      ''};
+    color: ${(props) =>
+      (props?.theme &&
+        Colors[props.theme as keyof ThemeColors]?.neutral?.neutral100) ||
+      ''};
+  }
+}`
+
+const Button = ({
```
Comment (gabrielcdiniz) [inline src/components/Button/index.tsx]: maybe needs specify type `React.FC`, for standardization as well as other components

File: src/components/TopBar/index.test.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { describe, it, expect } from 'vitest'
+
+describe('Renders main page correctly', async () => {
+  it('Should render the page correctly', async () => {
+    expect(true).equals(true)
+  })
+})
```
Comment (gabrielcdiniz) [inline src/components/TopBar/index.test.ts:7]: do we need this test file ?

File: src/components/Button/index.test.ts [hunk 7]
```diff
@@ -1,8 +1,7 @@
...
 describe('Renders main page correctly', async () => {
-    it('Should render the page correctly', async () => {
-        expect(true).equals(true);
-    });
-});
\ No newline at end of file
+  it('Should render the page correctly', async () => {
+    expect(true).equals(true)
+  })
+})
```
Comment (gabrielcdiniz) [inline src/components/Button/index.test.ts]: do we need this test file ?

File: src/types/index.ts (hunk 3 above)
Comment (gabrielcdiniz) [reply src/types/index.ts]: rename for "CommomProps" instead

**Review by gabrielcdiniz — CHANGES_REQUESTED** [review]

File: src/components/TopBar/index.test.ts (hunk 6 above)
Comment (samuelvermeuln) [reply src/components/TopBar/index.test.ts:7]: a task will need to be opened to write tests for these components

Both this and the other are just an initial example

File: src/components/Button/index.tsx (hunk 1 above)
Comment (samuelvermeuln) [reply src/components/Button/index.tsx:90]: In this case, if you remove webstorm, it complains, vscode lets it pass, but the IDE complains

File: src/components/TopBar/index.test.ts (hunk 6 above)
Comment (gabrielcdiniz) [reply src/components/TopBar/index.test.ts:7]: in my opinion, we remove and when we do the tests, we create. But we can go on like this

**Review by gabrielcdiniz — APPROVED** [review]

**Review by joelsalmeida — APPROVED** [review]

**Pull request merged**